src/components/agents/CommandStream.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <q-card-section class="q-px-xs q-pb-md q-pt-xs" v-if="hasText">
+    <script-output-copy-clip label="Live Output" :data="outputText" />
+    <q-separator class="q-my-sm" />
+  </q-card-section>
+  <div class="command-stream" ref="streamContainer">
+    <div class="terminal">
+      <pre class="mt-0">{{ outputText }}</pre>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, onUnmounted, watchEffect, nextTick, computed } from "vue";
+import { useAgentCmdWSConnection } from "@/websocket/agent";
+import ScriptOutputCopyClip from "@/components/scripts/ScriptOutputCopyClip.vue";
+import { uid } from 'quasar'
+
+const props = defineProps({
+  agentId: { type: String, required: true },
+  cmd: { type: String, required: true },
+  shell: { type: String, required: true },
+  timeout: { type: Number, default: 10 },
+});
+
+// emits
+const emit = defineEmits(["updateOutput", "streamLoaded", "streamClosed"]);
+
+const cmdId = uid();
+const { send, data, reset, close, status } = useAgentCmdWSConnection(props.agentId, cmdId);
+
+const outputText = ref("");
+const streamContainer = ref<HTMLElement | null>(null);
```

Comment:
useTemplateRef from vue is a nice shorthand for this. https://vuejs.org/guide/essentials/template-refs.html#accessing-the-refs